### PR TITLE
Fix confusing naming

### PR DESCRIPTION
**Pull request by HuKeping**
Signed-off-by: Hu Keping <hukeping@huawei.com>

### CODE REVIEW

File: client/changelist/change.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ const (
 // all changes in Snapshot and Timestamp are programmatically
 // generated base on Root and Targets changes.
 const (
-	TypeRootRole          = "role"
+	TypeRootRole          = "root"
```
Comment (ecordell) [inline client/changelist/change.go]: I agree that this is confusing naming, but if you look where this constant is used it looks like this is referring to "root role" as in "not-delegated" e.g. `root`, `snapshot`, `target`, and `timestamp`.

E.g. see https://github.com/docker/notary/blob/b4d4ab6c96e19c9f679f59270130966eb42fbec9/client/helpers.go#L176 , where it refers to any "base" role change. 

I think it makes more sense to change the constant to `TypeRole` or maybe `TypeBaseRole`

Comment (endophage) [inline client/changelist/change.go]: @ecordell is correct on the use case. Some more context, it's used specifically to describe a change type that modifies a base role, as opposed to a delegation role. Probably `TypeBaseRole` makes the most sense, to make it clear it's not to be used for delegation roles.

Comment (HuKeping) [inline client/changelist/change.go]: Thanks @ecordell @endophage! I'll check it later!

Comment (HuKeping) [inline client/changelist/change.go]: Updated :)

**Pull request merged**